packages/calypso-e2e/src/lib/pages/themes-page.ts

```diff
@@ -34,42 +24,52 @@ const selectors = {
  * @augments {BaseContainer}
  */
 export class ThemesPage extends BaseContainer {
-	/**
-	 * Constructs an instance of the component.
-	 *
-	 * @param {Page} page The underlying page.
-	 */
-	constructor( page: Page ) {
-		super( page );
-	}
-
 	/**
 	 * Post initialization steps.
 	 *
 	 * @returns {Promise<void>} No return value.
 	 */
 	async _postInit(): Promise< void > {
-		await this.page.waitForSelector( selectors.spinner, { state: 'hidden' } );
+		await Promise.all( [
+			this.page.waitForSelector( selectors.spinner, { state: 'hidden' } ),
+			this.page.waitForSelector( selectors.placeholder, { state: 'hidden' } ),
+			this.page.waitForLoadState( 'domcontentloaded' ),
+		] );
 	}
 
 	/**
-	 * Filters the themes on page by clicking on the selector passed in as argument.
+	 * Filters the themes on page by type.
 	 *
 	 * @param {string} type Pre-defined types of themes.
 	 * @returns {Promise<void>} No return value.
+	 * @throws {Error} If selected state of the theme filter button could not be confirmed.
 	 */
 	async filterThemes( type: 'All' | 'Free' | 'Premium' ): Promise< void > {
-		await this.page.click( selectors.showAllThemesButton );
-		const searchToolbar = await this.page.waitForSelector( selectors.searchToolbar );
+		const showAllThemesButton = await this.page.waitForSelector( selectors.showAllThemesButton, {
+			state: 'visible',
+		} );
+		await showAllThemesButton.waitForElementState( 'stable' );
+
+		// Click the 'Show all themes' button at bottom of gallery to expose the theme toolbar.
+		await showAllThemesButton.click();
+		const searchToolbar = await this.page.waitForSelector( selectors.searchToolbar, {
+			state: 'visible',
+		} );
+
+		// Retrive the target button.
 		const button = await searchToolbar.waitForSelector(
 			`a[data-e2e-value=${ type.toLowerCase() }]`
 		);
 		await button.click();
+
 		// This will wait for all placeholder classes to return to hidden state.
 		await this.page.waitForSelector( selectors.placeholder, { state: 'hidden' } );
+
 		// Verify that filter has been successfully applied.
 		const isSelected = await button.getAttribute( 'aria-checked' );
-		assert.strictEqual( isSelected, 'true' );
+		if ( ! isSelected ) {
+			throw new Error( `Failed to confirm the ${ type } button was clicked.` );
+		}
```

Comment:
Can we use the "load" state here as we did with the like button? The less we depend on selectors the better 👍 Something like:

```js
await buttonHandle.click();
await this.page.waitForFunction( button => button.ariaChecked === 'true', buttonHandle );
await this.page.waitForLoadState( 'load' );
```

Reply:
I tried it out and it seems to work locally. I will push up a new revision and run it in TC to verify.

I did modify the ordering of the calls a little bit:
```
		await button.click();
		await this.page.waitForFunction( ( element: any ) => element.ariaChecked === 'true', button );
```

It looks like (at least locally) the `waitForLoadState` was not necessary for successful run of the test.